priam/src/main/java/com/netflix/priam/aws/S3FileSystemBase.java

```diff
@@ -121,42 +129,77 @@ public void cleanup() {
         }
     }
 
-    private boolean updateLifecycleRule(IConfiguration config, List<Rule> rules, String prefix) {
-        Rule rule = null;
-        for (BucketLifecycleConfiguration.Rule lcRule : rules) {
-            if (lcRule.getPrefix().equals(prefix)) {
-                rule = lcRule;
-                break;
+    // Dummy class to get Prefix. - Why oh why AWS you can't give the details!!
+    private class PrefixVisitor implements LifecyclePredicateVisitor {
```

Comment:
Think this should either be static or an anonymous class (something something memory leaks)